server/routes/appointments/appointmentsController.ts

```diff
@@ -183,32 +184,41 @@ export default class AppointmentsController {
     if (draft.data === null) {
       throw new Error('Draft data was unexpectedly null when submitting appointment')
     }
-
-    try {
-      await this.interventionsService.scheduleSupplierAssessmentAppointment(
-        res.locals.user.token.accessToken,
-        supplierAssessment.id,
-        draft.data
+    const appointmentScheduleDetails = this.extractAppointmentSchedulingDetails(fetchResult.draft)
+    if (appointmentScheduleDetails === null || !appointmentScheduleDetails.appointmentTime) {
+      throw new Error('No scheduling details were filled in on the draft')
+    }
+    if (this.isAPastAppointment(appointmentScheduleDetails)) {
+      res.redirect(
+        `/service-provider/referrals/${req.params.id}/supplier-assessment/post-assessment-feedback/edit/${draft.id}/attendance`
       )
```

Comment:
can we explicitly return after the redirect, instead of relying on this conditional being the last thing in the method?

Reply:
I added some returns after seeing your previous bug fix but the linter kept saying they weren't needed and automatically removed them when I added.

Reply:
done